sgx/ias-types/src/parsed.rs

```diff
@@ -0,0 +1,392 @@
+
+    /// The quote body minus the signature.
+    pub quote: Quote,
+
+    /// An optional string explaining the quote revocation, if quote_error is GroupRevoked.
+    pub revocation_reason: Option<RevocationCause>,
+
+    /// An optional error used to indicate the results of IAS checking the PSE manifest.
+    pub pse_manifest_status: Option<PseManifestResult>,
+
+    /// The hash of the PSE manifest, if provided.
+    pub pse_manifest_hash: Option<Vec<u8>>,
+
+    /// PSW-provided opaque data.
+    pub platform_info: Option<PlatformInfo>,
+
+    /// The IAS request nonce.
+    pub nonce: Option<Nonce>,
+
+    /// A unique hardware ID returned when a linkable quote is requested.
+    pub epid_pseudonym: Option<EpidPseudonym>,
+
+    /// The String URL of an Intel security advisory page.
+    pub advisory_url: Option<String>,
+
+    /// A set of advisory identifier strings applicable to this platform, as returned by IAS
+    pub advisory_ids: BTreeSet<String>,
+}
+
+impl<'src> TryFrom<&'src IasReport> for Report {
+    type Error = Error;
+
+    /// Parse the JSON contents of a VerificationReport into a
+    /// VerificationReportData object
+    fn try_from(src: &'src IasReport) -> Result<Self, Error> {
+        // Parse the JSON into a hashmap
+        let (chars_parsed, data) = parse(&src.http_body);
+        if data.is_none() {
+            return Err(JsonError::NoData.into());
+        }
+
+        if chars_parsed < src.http_body.len() {
+            return Err(JsonError::IncompleteParse(chars_parsed).into());
+        }
+
+        let mut data = match data.unwrap() {
+            Value::Object(o) => o,
+            _ => return Err(JsonError::RootNotObject.into()),
+        };
+
+        // Actually parse the JSON into real fields
+        let id = data
+            .remove("id")
+            .ok_or_else(|| JsonError::FieldMissing("id".to_owned()))?
+            .try_into()?;
+        let string_timestamp: String = data
+            .remove("timestamp")
+            .ok_or_else(|| JsonError::FieldMissing("timestamp".to_owned()))?
+            .try_into()?;
+        let naive_timestamp = NaiveDateTime::parse_from_str(&string_timestamp, "%Y-%m-%d %H:%M:%S")
+            .map_err(Error::Timestamp)?;
+        let timestamp = DateTime::from_utc(naive_timestamp, Utc);
+        let version = data
+            .remove("version")
+            .ok_or_else(|| JsonError::FieldMissing("version".to_owned()))?
+            .try_into()?;
+
+        // Get the PIB, used in QuoteError, PseManifestError
+        let platform_info = match data.remove("platformInfoBlob") {
+            Some(v) => {
+                let value: String = v.try_into()?;
+                Some(PlatformInfo::from_hex(&value).map_err(Error::PlatformInfo)?)
+            }
+            None => None,
+        };
+
+        // Get the (optional) revocation reason, used in QuoteError
+        let revocation_reason = match data.remove("revocationReason") {
```

Comment:
Nit: `.map()` instead of the match (relevant in other places as well)?

Reply:
Ah, yeah, thanks---this code was written before `Option::transpose()` was stabilized.